File: ordenação/JavaScript/insertion_sort.js
let vetor = [5, 3, 8, 4, 2];

// Começa no segundo elemento
for (let i = 1; i < vetor.length; i++) {

    // Guarda o elemento atual
    let chave = vetor[i];
    let j = i - 1;

    // Move elementos maiores para a direita
    while (j >= 0 && vetor[j] > chave) {
        vetor[j + 1] = vetor[j];
        j--;
    }

    // Coloca o elemento na posição correta
    vetor[j + 1] = chave;
}

console.log("Vetor ordenado:", vetor);

File: ordenação/Python/insertion_sort.py
vetor = [5, 3, 8, 4, 2]

# Começa no segundo elemento
for i in range(1, len(vetor)):

    # Guarda o elemento atual
    chave = vetor[i]
    j = i - 1

    # Move elementos maiores para a direita
    while j >= 0 and vetor[j] > chave:
        vetor[j + 1] = vetor[j]
        j -= 1

    # Coloca a chave na posição correta
    vetor[j + 1] = chave

print("Vetor ordenado:", vetor)
